### PR TITLE
prepare for CIDR addresses

### PR DESCRIPTION
**Pull request by zealws**
TLDR; lots of cleanup to make code in `src/shaping` ready for CIDR addresses.
## Abstracted out `net.IP` usage in `src/shaping` to allow for CIDR addresses in the future.

The idea here is `Shaper`s will accept values of type `Target` rather than `net.IP`. An implementation of these target types can be found in `src/iptables/targets.go`. Clients are responsible for converting `net.IP` or `net.IPNet` values into the appropriate Target type before passing them to the `Shaper`.
Currently `atcd` only allows `IPTarget` values.

I'm not quite sure what effect the inclusion of CIDR address will do to the thrift spec, but this approach to the `Shaper` should give us the freedom to make changes to the thrift spec easily.
## Added `src/iptables` which has simple general-purpose iptables bindings.

The code is self-contained in `src/iptables` and has no dependencies outside the go standard library.

There's a lot of stuff here, so if you want we can discuss this in person to go over fine details before landing.

### CODE REVIEW

**Comment by zfjagann** [comment]
cidr

**Comment by chantra** [comment]
Can you also add unittest for ipv6?

**Comment by zfjagann** [comment]
Yeah sure. I'll do this later tonight.

**Comment by zfjagann** [comment]
Good call on the unit tests for IPv6, they flushed out some differences between ip6tables and iptables that would have been an issue in the future.

**Comment by chantra** [comment]
:dog:
